src/Framework/StringBuilderCache.cs

```diff
@@ -31,17 +31,21 @@
 **
 ===========================================================*/
 
+//#define ASSERT_BALANCE
+
 using System;
+using System.Diagnostics;
 using System.Text;
+#if !CLR2COMPATIBILITY && !MICROSOFT_BUILD_ENGINE_OM_UNITTESTS
+using Microsoft.Build.Eventing;
+#endif
 
-namespace Microsoft.Build.Shared
+namespace Microsoft.Build.Framework
 {
     internal static class StringBuilderCache
     {
-        // The value 360 was chosen in discussion with performance experts as a compromise between using
-        // as little memory (per thread) as possible and still covering a large part of short-lived
-        // StringBuilder creations on the startup path of VS designers.
-        private const int MAX_BUILDER_SIZE = 360;
+        // The value 512 was chosen empirically as 95% percentile
```

Comment:
95%ile of what?

Reply:
![image](https://user-images.githubusercontent.com/12969783/144148806-0d4f5b65-d806-4b06-9f96-dfd2f7d2df82.png)
(From https://github.com/dotnet/msbuild/issues/2697#issuecomment-981140614)

Reply:
"Bellow is "histogram" of ReuseableStringBuilder returning length. Y axis is sum of returning length for given bucket."

Reply:
Changed to:
`// The value 512 was chosen empirically as 95% percentile of returning string length.`